packages/ag-charts-community/src/chart/series/sankey/sankeySeries.ts

```diff
@@ -1,29 +1,391 @@
 import type { ModuleContext } from '../../../module/moduleContext';
 import type { BBox } from '../../../scene/bbox';
+import { Group } from '../../../scene/group';
+import { Selection } from '../../../scene/selection';
+import { Rect } from '../../../scene/shape/rect';
 import type { PointLabelDatum } from '../../../scene/util/labelPlacement';
+import { ARRAY, Validate } from '../../../util/validation';
 import type { ChartAnimationPhase } from '../../chartAnimationPhase';
 import type { ChartAxisDirection } from '../../chartAxisDirection';
-import type { DataController } from '../../data/dataController';
+import { DataController } from '../../data/dataController';
+import type { DataModel, ProcessedData } from '../../data/dataModel';
+import { createDatumId } from '../../data/processors';
 import type { TooltipContent } from '../../tooltip/tooltip';
-import { Series, type SeriesNodeDataContext } from '../series';
+import { DataModelSeries } from '../dataModelSeries';
+import type { FlowProportionSeries } from '../flowProportionSeries';
+import { type PickFocusInputs, type SeriesNodeDataContext, keyProperty, valueProperty } from '../series';
+import type { SeriesNodeDatum } from '../seriesTypes';
+import { SankeyLink } from './sankeyLink';
 import { SankeySeriesProperties } from './sankeySeriesProperties';
+import { type NodePathLengthEntry, computePathLength } from './sankeyUtil';
+
+interface SankeyLinkDatum extends SeriesNodeDatum {
+    fromId: string;
+    toId: string;
+    size: number;
+    x1: number;
+    x2: number;
+    y1: number;
+    y2: number;
+    height: number;
+}
+
+interface SankeyNodeDatum {
+    id: string;
+    label: string | undefined;
+    size: number;
+    x: number;
+    y: number;
+    width: number;
+    height: number;
+}
+
+interface SankeyNodeLabelDatum {}
+
+export interface SankeyNodeDataContext extends SeriesNodeDataContext<SankeyLinkDatum, SankeyNodeLabelDatum> {
+    nodes: SankeyNodeDatum[];
+}
+
+export class SankeySeries
+    extends DataModelSeries<SankeyLinkDatum, SankeySeriesProperties, SankeyNodeLabelDatum, SankeyNodeDataContext>
+    implements FlowProportionSeries
+{
+    static readonly className = 'SankeySeries';
+    static readonly type = 'sankey' as const;
 
-export class SankeySeries extends Series<any, any> {
     override properties = new SankeySeriesProperties();
 
+    @Validate(ARRAY, { optional: true, property: 'nodes' })
+    private _chartNodes?: any[] = undefined;
+
+    private get nodes() {
+        return this.properties.nodes ?? this._chartNodes;
+    }
+
+    private readonly nodesDataController = new DataController('standalone');
```

Comment:
Might need help with this one. I get close then it can't resolve the properties. The scope cache appears to be empty - but the resolved data seems to be right
<img width="785" alt="image" src="https://github.com/ag-grid/ag-charts/assets/7275322/a8c8ad2f-f858-4954-876d-22c1bfbad7d2">

Reply:
Discussed offline, something is awry so let's continue with the previous approach